raft.go

```diff
@@ -1080,10 +1103,14 @@ func (r *raft) Step(m pb.Message) error {
 
 	switch m.Type {
 	case pb.MsgHup:
+		force := bytes.Equal(m.Context, forceCampaign)
+		if force && r.readOnly.option == ReadOnlyLeaseBased {
+			r.logger.Panic("can't force an election with ReadOnlyLeaseBased")
```

Comment:
Why panic? We can "just" ignore the MsgHup, can't we?

Reply:
We can, but this is a programming error. The caller should have checked this already and returned an error (see `CampaignNow()`). I'm fine with downgrading this to an error and dropping the message though.